rest/context.go

```diff
@@ -325,6 +329,30 @@ func (ctx *gorillaRequestContext) NextURL() (string, error) {
 	return u.String(), nil
 }
 
+func (ctx *gorillaRequestContext) BuildURL(routeName, resourceID string) (string, error) {
+	r, ok := ctx.Request()
+	if !ok {
+		return "", fmt.Errorf("Unable to build next url: no request")
+	}
+
+	scheme := "http"
+	if r.TLS != nil {
+		scheme += "s"
+	}
+
+	// Need a way to get route name without knowledge of go-rest internals
+	// routeName := resourceName + ":read"
+	uri, err := ctx.router.Get(routeName).
+		Schemes(scheme).
+		Host(r.Host).
+		URL("version", ctx.Version(),
+		"resource_id", resourceID)
+	if err != nil {
+		log.Println(err)
+	}
+	return fmt.Sprintf("%v", uri), nil
```

Comment:
`Sprintf` is expensive because it has to parse the formatting string and then apply reflect to its arguments. It's not doing anything here anyway. Instead, call `uri.String()` directly.